src/App.tsx

```diff
@@ -1,24 +1,107 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import { UserWarning } from './UserWarning';
+import { HeaderTodoApp } from './components/HeaderTodoApp';
+import { MainTodoApp } from './components/MainTodoApp';
+import {
+  addTodo, deleteTodo, getTodos,
+} from './api/todos';
+import { Todo } from './types/Todo';
+import { FooterTodoApp } from './components/FooterTodoApp';
+import { Filters } from './types/Filters';
+import { ErrorComponent } from './components/ErrorComponent';
 
-const USER_ID = 0;
+const USER_ID = 10299;
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [category, setCategory] = useState<Filters>(Filters.All);
+  const [tempTodo, setTempTodo] = useState<Todo | null>(null);
+  const [error, setError] = useState('');
+
+  const loadTodos = useCallback(async () => {
+    const todosFromServer = await getTodos(USER_ID);
+
+    setTodos(todosFromServer);
+    setTempTodo(null);
+  }, []);
+
+  useEffect(() => {
+    loadTodos();
+  }, []);
+
+  const visibleTodos = useMemo(() => todos.filter(({ completed }) => {
+    switch (category) {
+      case Filters.completed:
+        return completed === true;
+      case Filters.Active:
+        return completed === false;
+      default:
+        return true;
+    }
+  }), [todos, category]);
+
+  const createTodo = useCallback(async (todoData: Todo) => {
+    try {
+      setTempTodo(todoData);
+      await addTodo(todoData);
+    } catch {
+      setError('Unable to add a todo');
+      setTempTodo(null);
+    }
+
+    loadTodos();
+  }, []);
+
+  const removeTodo = useCallback(async (todoData: Todo) => {
+    try {
+      setTempTodo(todoData);
+      await deleteTodo(todoData.id);
+    } catch {
+      setError('Unable to delete a todo');
+      setTempTodo(null);
+    }
+
+    loadTodos();
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-loading-todos#react-todo-app-load-todos">React Todo App - Load Todos</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <HeaderTodoApp
+          todos={todos}
+          USER_ID={USER_ID}
```

Comment:
why you put it in the props?
It's constant, just move `USER_ID` to the separate file & import it